src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -509,31 +552,39 @@ public bool TryUpdate(TKey key, TValue newValue, TValue comparisonValue)
         /// </summary>
         /// <param name="key">The key whose value is compared with <paramref name="comparisonValue"/> and
         /// possibly replaced.</param>
-        /// <param name="hashcode">The hashcode computed for <paramref name="key"/>.</param>
+        /// <param name="nullableHashcode">The hashcode computed for <paramref name="key"/>.</param>
         /// <param name="newValue">The value that replaces the value of the element with <paramref
         /// name="key"/> if the comparison results in equality.</param>
         /// <param name="comparisonValue">The value that is compared to the value of the element with
         /// <paramref name="key"/>.</param>
-        /// <returns>true if the value with <paramref name="key"/> was equal to <paramref
-        /// name="comparisonValue"/> and replaced with <paramref name="newValue"/>; otherwise,
-        /// false.</returns>
-        /// <exception cref="System.ArgumentNullException"><paramref name="key"/> is a null
-        /// reference.</exception>
-        private bool TryUpdateInternal(TKey key, int hashcode, TValue newValue, TValue comparisonValue)
+        /// <returns>
+        /// true if the value with <paramref name="key"/> was equal to <paramref name="comparisonValue"/> and
+        /// replaced with <paramref name="newValue"/>; otherwise, false.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is a null reference.</exception>
+        private bool TryUpdateInternal(TKey key, uint? nullableHashcode, TValue newValue, TValue comparisonValue)
```

Comment:
`uint?` requires an extra register to be passed on 32-bit platforms and there is only one call site passing `null`. Could we keep `uint hashcode`?

Reply:
Though if that would break inlinability of `TryUpdate`, keeping nullable is OK.